src/ui/settings/__tests__/index.js

```diff
@@ -0,0 +1,307 @@
+import sinon from 'sinon';
+import expect from 'expect.js';
+import init from '..';
+import defaultsProvider from '../defaults';
+
+describe('ui settings', function () {
+  describe('overview', function () {
+    it('has expected api surface', function () {
+      const { uiSettings } = instantiate();
+      expect(typeof uiSettings.get).to.be('function');
+      expect(typeof uiSettings.getAll).to.be('function');
+      expect(typeof uiSettings.getDefaults).to.be('function');
+      expect(typeof uiSettings.getRaw).to.be('function');
+      expect(typeof uiSettings.getUserProvided).to.be('function');
+      expect(typeof uiSettings.remove).to.be('function');
+      expect(typeof uiSettings.removeMany).to.be('function');
+      expect(typeof uiSettings.set).to.be('function');
+      expect(typeof uiSettings.setMany).to.be('function');
+    });
+  });
+
+  describe('#setMany()', function () {
+    it('returns a promise', () => {
+      const { uiSettings } = instantiate();
+      const result = uiSettings.setMany({ a: 'b' });
+      expect(typeof result.then).to.be('function');
```

Comment:
Do we know what promise implementation is returned? If so, expect.js's constructor assertion could be used instead of duck typing:

`expect(result).to.be.a(Promise);`